File: Python/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        i = 0
        for num in nums:
            if target - num in dict:
                return [i, dict[target - num]]
            dict[num] = i
            i += 1

        

File: Java/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ret = new int[2];
        Map<Integer, Integer> result = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            if (result.containsKey(target - nums[i])) {
                return new int[]{result.get(target - nums[i]), i};
            }
            result.put(nums[i], i);
        }         

        return new int[]{};
    }
}
